src/notebooks/controllers/notebookControllerManager.ts

```diff
@@ -423,7 +428,7 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
             }
         });
 
-        return defaultPython3Kernel || defaultPythonKernel || defaultPythonLanguageKernel || controllers[0];
```

Comment:
We were returning a random controller, that was causing even more issues 
Basically, why aren't we using the active interpreter like all other cases.

Reply:
Good catch, I think I ran into this once recently, and it was confusing.